javascripts/discourse/connectors/before-main-outlet/right-blocks.gjs

```diff
@@ -10,22 +10,19 @@ import BlockProfile from "../../components/blocks/profile";
 import BlockTime from "../../components/blocks/time";
 import BlockTopContributors from "../../components/blocks/top-contributors";
 import BlockTopTopics from "../../components/blocks/top-topics";
-import StickySidebarComponent from "../../components/sticky-sidebar";
+import StickySidebar from "../../components/sticky-sidebar";
 
 export default class RightBlocks extends Component {
-  @service currentUser;
   @service router;
-  @service site;
-  @service discovery;
 
   get blocks() {
     return settings.blocks;
   }
 
   get shouldRenderBlocks() {
     return (
-      this.router?.currentRoute?.parent?.name === "discovery" &&
-      this.router?.currentRouteName !== "discovery.categories"
+      this.router.currentRoute.parent?.name === "discovery" &&
```

Comment:
is `router.currentRoute` (RouteInfo) guaranteed to be present?

Reply:
According to the code/jsdoc, yes, it's non null: (and in most places we already assume it's present 😃)

https://github.com/emberjs/ember.js/blob/4671d467f57c83dd8bc6f40c310df1c8c4b86b42/packages/%40ember/routing/router-service.ts#L747